### ISSUE TITLE
pnpm workspace not work

### ISSUE THREAD

**Issue by ckken** [parcel-bundler/parcel#7063]
![image](https://user-images.githubusercontent.com/3890513/137075426-5274a101-40f8-4d64-b017-74b5f122b57b.png)

**Comment by ckken** [comment]
Mark

**Comment by ch99q** [comment]
I've experienced the same thing, but I have been investigating a bit into this.

I found two things resulting in this.
1. It won't identify the folder you have your project in cause it won't have a lock file.
https://github.com/parcel-bundler/parcel/blob/d2a9b6bdd93855b468fa9911dad27a49e54e9557/packages/core/core/src/resolveOptions.js#L52-L59
2. If we add package.json or "node_modules" we return to issue 2; it won't necessarily find the root folder cause of the entries. So to make it work now, we have to do `parcel build src/index.ts`; this prevents us from using `"source": "src/index.ts"` in the `package.json`.

I've gone so deep into the code; I can't figure where it comes from; To me, it seems like it doesn't handle the `"source"` in the beginning but first later in the process. @devongovett can probably give some good insight into this.
